test/test_c10d.py

```diff
@@ -509,10 +515,103 @@ def test_gloo_backend(self):
+            # filled with the tensors from the master.
+            tensors = torch.zeros(10, device=device).chunk(2)
+
+        c10d._dist_broadcast_coalesced(
+            tensors,
+            buffer_size=25 * MB,
+            process_group=process_group)
+
+        if not self.is_master:
+            self.assertEqual(tensors, target)
+
+    @skip_if_not_multigpu
+    def test_sync_params_no_buffers(self):
+        # Set up process group.
+        store = c10d.TCPStore('localhost', self.port, self.is_master)
+        options = c10d.ProcessGroupGloo.Options()
+        options.devices = [c10d.ProcessGroupGloo.create_tcp_device(interface="lo")]
+        process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size, options)
+
+        # Use all available devices on every process here (data is small, so should be fine).
+        devices = list(range(torch.cuda.device_count()))
+        target = torch.arange(10, dtype=torch.float64, device='cuda:0').chunk(2)
+        parameter_data = [target]
+        parameter_data += [torch.zeros(10, device=torch.device('cuda', d)).chunk(2) for d in devices[1:]]
+        buffer_data = [[]] * len(parameter_data)
+
+        c10d._sync_params(
+            process_group,
+            parameter_data=parameter_data,
+            buffer_data=buffer_data,
+            devices=devices,
+            broadcast_bucket_size=25 * MB,
+            broadcast_buffers=False)
+
+        for device_data in parameter_data:
+            for i, parameter in enumerate(device_data):
+                self.assertEqual(parameter, target[i])
+
+    @skip_if_not_multigpu
+    def test_sync_params_with_buffers(self):
+        # Set up process group.
+        store = c10d.TCPStore('localhost', self.port, self.is_master)
+        options = c10d.ProcessGroupGloo.Options()
+        options.devices = [c10d.ProcessGroupGloo.create_tcp_device(interface="lo")]
+        process_group = c10d.ProcessGroupGloo(store, self.rank, self.world_size, options)
+
+        devices = list(range(torch.cuda.device_count()))
+        target = torch.arange(10, dtype=torch.float64, device='cuda:0').chunk(2)
+        parameter_data = [target]
+        parameter_data += [torch.zeros(10, device=torch.device('cuda', d)).chunk(2) for d in devices[1:]]
+
+        # sync_params should do a dist_broadcast for buffers, so we only populate the master buffers and
+        # then check that other processes' tensors end up matching.
+
+        if self.is_master:
+            buffer_data = [target]
+            buffer_data += [torch.zeros(10, device=torch.device('cuda', d)).chunk(2) for d in devices[1:]]
+        else:
+            buffer_data = [torch.zeros(10, device=torch.device('cuda', d)).chunk(2) for d in devices]
+
+        c10d._sync_params(
+            process_group,
+            parameter_data=parameter_data,
+            buffer_data=buffer_data,
+            devices=devices,
+            broadcast_bucket_size=25 * MB,
```

Comment:
will 25MB be too big in this case? Ideally, we would like to have more than 1 bucket to be created.

Reply:
Good point